packages/number/src/NumberEditor.tsx

```diff
@@ -26,37 +30,80 @@ function getRangeFromField(field: FieldAPI): RangeValidation {
   return result ? result.range : {};
 }
 
-export function NumberEditor(props: NumberEditorProps) {
-  const { field } = props;
+function valueToString(value: InnerNumberEditorProps['value']) {
+  return value === undefined ? '' : String(value);
+}
+
+type InnerNumberEditorProps = Pick<
+  FieldConnectorChildProps<number>,
+  'disabled' | 'errors' | 'setValue' | 'value'
+> & {
+  field: NumberEditorProps['field'];
+};
 
+function InnerNumberEditor({ disabled, errors, field, setValue, value }: InnerNumberEditorProps) {
+  const previousValue = React.useRef(valueToString(value));
+  const [stringValue, setStringValue] = React.useState(valueToString(value));
   const range = getRangeFromField(field);
 
+  React.useEffect(() => {
+    previousValue.current = valueToString(value);
+  }, [value]);
+
+  React.useEffect(() => {
+    if (previousValue.current !== String(value) && String(value) !== stringValue) {
+      setStringValue(valueToString(value));
+    }
+  }, [stringValue, value]);
```

Comment:
Can you elaborate on this logic pls. Based on code it seems like previousValue.current will be always equal to String(value), because you are updating it on every value change in the useEffect above. Isn't it?

Reply:
I've updated with some hopefully more clear variable names and a comment which I hope helps.

The first effect is to set a reference that updates when the value converted to a string is changed. This is needed for the second effect, where we check if the value sent from field connector has changed. If it has we can assume that it's valid and then we should update the input value to show it. The comparison with the reference is necessary because the second effect also runs when the input value changes, and in those cases we don't want to trigger the call to update the input value.

Reply:
Writing it out like that I see that we could perhaps get the samme outcome from not listing the input value in the effect's dependency array, although that would probably trigger a lint warning.